src/lightning/data/streaming/data_processor.py

```diff
@@ -101,6 +101,16 @@ def _wait_for_file_to_exist(s3: S3Client, obj: parse.ParseResult, sleep_time: in
                 raise e
 
 
+def _wait_for_disk_usage_higher_than_threshold(threshold_in_gb: int = 25, sleep_time: int = 3) -> None:
+    usage = shutil.disk_usage("/")
+
+    while (usage.free / 1000 / 1000 / 1000) <= threshold_in_gb:
```

Comment:
```suggestion
    while (usage.free / 1024 / 1024 / 1024) <= threshold_in_gb:
```
ref: https://www.telecomhall.net/t/convert-kilobits-kbit-to-megabytes-mb-1000-or-1024-r/13194/4

Reply:
but the threshold and the usage.free is in bytes, not in bibytes. This would be wrong.

Reply:
It is fine. I am using GB.